File: update_globals.js
const fs = require('fs');
const path = require('path');

function updateGlobalElements() {
    // Read the source of truth (index.html)
    const indexContent = fs.readFileSync('index.html', 'utf-8');

    // Extract header block
    const headerMatch = indexContent.match(/<!-- Main Header-->[\s\S]*?<!--End Main Header -->/);
    if (!headerMatch) {
        console.error("Error: Could not find header in index.html");
        return;
    }
    const headerContent = headerMatch[0];

    // Extract footer block
    const footerMatch = indexContent.match(/<footer[\s\S]*?<\/footer>/);
    if (!footerMatch) {
        console.error("Error: Could not find footer in index.html");
        return;
    }
    const footerContent = footerMatch[0];

    const files = fs.readdirSync('.');
    const htmlFiles = files.filter(f => f.endsWith('.html') && f !== 'index.html');
    
    let count = 0;
    for (const file of htmlFiles) {
        let content = fs.readFileSync(file, 'utf-8');
        let modified = false;
        
        // Replace header
        if (content.includes('<!-- Main Header-->')) {
            content = content.replace(/<!-- Main Header-->[\s\S]*?<!--End Main Header -->/, headerContent);
            modified = true;
        } else {
            // Fallback
            const replaced = content.replace(/<header[\s\S]*?<\/header>/, headerContent);
            if (replaced !== content) {
                content = replaced;
                modified = true;
            }
        }
                
        // Replace footer
        const replacedFooter = content.replace(/<footer[\s\S]*?<\/footer>/, footerContent);
        if (replacedFooter !== content) {
            content = replacedFooter;
            modified = true;
        }
            
        if (modified) {
            fs.writeFileSync(file, content, 'utf-8');
            count++;
            console.log(`Updated ${file}`);
        }
    }

    console.log(`Total files updated: ${count}`);
}

updateGlobalElements();

File: update_globals.py
import os
import re

def update_global_elements():
    # Read the source of truth (index.html)
    with open('index.html', 'r', encoding='utf-8') as f:
        index_content = f.read()

    # Extract header block
    header_match = re.search(r'<!-- Main Header-->.*?<!--End Main Header -->', index_content, flags=re.DOTALL)
    if not header_match:
        print("Error: Could not find header in index.html")
        return
    header_content = header_match.group(0)

    # Extract footer block
    footer_match = re.search(r'<footer.*?>.*?</footer>', index_content, flags=re.DOTALL)
    if not footer_match:
        print("Error: Could not find footer in index.html")
        return
    footer_content = footer_match.group(0)

    html_files = [f for f in os.listdir('.') if f.endswith('.html') and f != 'index.html']
    
    count = 0
    for file in html_files:
        with open(file, 'r', encoding='utf-8') as f:
            content = f.read()
            
        modified = False
        
        # Replace header
        # Some files might not have <!-- Main Header-->, they might just have <header>
        if '<!-- Main Header-->' in content:
            content = re.sub(r'<!-- Main Header-->.*?<!--End Main Header -->', header_content, content, flags=re.DOTALL)
            modified = True
        else:
            # Fallback for files without the comment wrappers
            content, n = re.subn(r'<header.*?>.*?</header>', header_content, content, flags=re.DOTALL)
            if n > 0:
                modified = True
                
        # Replace footer
        content, n = re.subn(r'<footer.*?>.*?</footer>', footer_content, content, flags=re.DOTALL)
        if n > 0:
            modified = True
            
        if modified:
            with open(file, 'w', encoding='utf-8') as f:
                f.write(content)
            count += 1
            print(f"Updated {file}")

    print(f"Total files updated: {count}")

if __name__ == '__main__':
    update_global_elements()
